src/main/java/net/minestom/server/event/player/PlayerCheckPermissionEvent.java

```diff
@@ -0,0 +1,42 @@
+package net.minestom.server.event.player;
+
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.trait.PlayerEvent;
+import net.minestom.server.permission.Permission;
+import org.jetbrains.annotations.NotNull;
+
+public class PlayerCheckPermissionEvent implements PlayerEvent {
+
+    private final Player player;
+    private final Permission permission;
+    private final String permissionName;
```

Comment:
Why do you need both the permission as well as its name? You can just get the name from the Permission object easily.